File: functions/ffmpeg/variants/go/0_flac_get_max_frame_size_3.go
package main

func flacGetMaxFrameSize(blocksize, ch, bps int32) int32 {
    base := int32(16)
    perChannel := (7 + bps + 7) / 8
    frameSize := int32(0)
    if ch == 2 {
        frameSize = ((2*bps + 1) * blocksize + 7) / 8
    } else {
        frameSize = (ch * bps * blocksize + 7) / 8
    }
    count := base + ch*perChannel + frameSize + 2
    return count
}


func main(){
    flacGetMaxFrameSize(0,0,0)
}

File: functions/ffmpeg/variants/c/0_flac_get_max_frame_size_3.c
#include <stdint.h>

static int flac_get_max_frame_size(int blocksize, int ch, int bps) 
{
    int count = 16;
    count += ch * ((bps + 14) / 8);
    
    int size;
    if (ch == 2) {
        size = ((2 * bps + 1) * blocksize + 7) / 8;
    } else {
        size = (ch * bps * blocksize + 7) / 8;
    }
    count += size;
    
    count += 2;
    
    return count;
}


int main(){
    flac_get_max_frame_size(0,0,0);
}
